core/src/main/java/org/infinispan/commands/functional/TxReadOnlyKeyCommand.java

```diff
@@ -59,11 +59,22 @@ public void readFrom(ObjectInput input) throws IOException, ClassNotFoundExcepti
    }
 
    @Override
+   public void init(ComponentRegistry componentRegistry) {
+      super.init(componentRegistry);
+      // This may be called from parent's constructor when mutations are not initialized yet
```

Comment:
Sorry what implications does this then have? It seems the `mutations` collection is not thread safe, so can we get `ConcurrentModificationException`?

Reply:
@wburns I think it's just that on remote nodes the mutations list is initialized before `init()`, and on on the originator the mutations are added (already initialized) after `init()`. From a thread-safety perspective, AFAICT the collection is unique the command, and we never have 2 threads accessing the same command -- or we have a bug.

Reply:
Mr. Berindei is correct, it's just about initialization order, we need the `if`.